torch/_inductor/codegen/cpp.py

```diff
@@ -646,7 +646,17 @@ def tanh(x):
 
     @staticmethod
     def signbit(x):
-        return f"std::signbit({x})"
+        """
+        On windows std::signbit only support float type.
+        Ref: https://learn.microsoft.com/en-us/cpp/c-runtime-library/reference/signbit?view=msvc-170
+        So, we need to cast type to `long double` for accuracy.
+        MSVC only support 64-bit precision, clang is support 80-bits precision.
+        """
+        return (
+            f"std::signbit(static_cast<long double>({x}))"
```

Comment:
I don't think we need `long double` here, since signbit is just checking positive/negative.  `float` should be fine.

Reply:
Done.